src/smt/theory_str_noodler/theory_str_noodler.h

```diff
@@ -258,6 +270,12 @@ namespace smt::noodler {
          * @param name Name of the var
          */
         expr_ref mk_int_var(const std::string& name) {
+            // quantified int variables we need to create as z3 variables. If they are created as skolem const, the quantification is 
+            // ignored (because everything there is a constant).
+            auto it = this->quantif_vars.find(name);
+            if(it != this->quantif_vars.end()) {
+                return expr_ref(m.mk_var(it->second, m_util_a.mk_int()), m);
+            }
```

Comment:
Is it possible that we will both have quantified and unquantified variable with the same name? Or does Z3 take care of it? Because then this could be problematic.

Reply:
As a matter of fact the names of the bound variables do not matter much. Z3 represents the bound variable using deBruijn's indices (basically bound variables are identified with numbers).

Reply:
So is this function never called for quantified variables?

Reply:
Yes, if you have multiple occurrences of a variable with the same name in different scopes, it may cause problems (not occurred now, we were aware of it). I leave a comment to make it clear.